### ISSUE TITLE
DBs and Tables are not updating after create

### ISSUE THREAD

**Issue by cperezabo** [sqlectron/sqlectron-gui#134]
When you run a create database or create table, the created object doesn't show at the sidebar, you have to reload the complete app!

**Comment by BornaP** [comment]
I think there is no database gui client (i.e. MySQL Workbench, pgAdmin) which shows newly created database/table without manually clicking on reloading data, but I agree we can try to implement this. Or at least put 'refresh button' which would reload databases and tables (like other clients do)

**Comment by maxcnunes** [comment]
Even the auto refresh looking a best solution. We can not implement that because there is not a properly way to detect the query type in nodejs. Regex is not a good option and the best solution would be using AST of the query (which there is not any good one available in nodejs - is the same problem described [here](https://github.com/sqlectron/sqlectron-gui/issues/129#issuecomment-216000135)).
So the current best and simple solution is including a refresh button.

**Comment by darkwisperer** [comment]
Random thought, on sql execute IF SQL string contains keyword "Create" then run the refresh menu function. I might be oversimplifying the issue.

i.e
`if ( string.indexOf('create') > -1 ||  string.indexOf('Create') > -1)
      refreshLeftMenu();`

**Comment by maxcnunes** [comment]
Sorry. In this case there is no simple solution. Your suggestions would not work with many situations 😢 
```sql
INSERT INTO mytable VALUES ('I have created something today');
```
or
```sql
-- query created at 11/09/2011
SELECT * FROM mytable;
```
or
```sql
SELECT id, name, createdAt FROM people;
```
or
```sql
SELECT * FROM creations;
```
These are only the few simple cases I have just thought about right now. I'm sure there are others even more complex that a string "search" or even Regex would not work. Only a AST of the query would give us the full understanding of the query to be able to "react" over the content.

**Comment by darkwisperer** [comment]
Your right. My idea would have unnecessarily done a refresh. I noticed there are a few AST related npm packages that we might be able to leverage. This is a little over my head, so I'm just going to follow along :)

**Comment by cperezabo** [comment]
@BornaP oh yes it is. I use Sequel Pro for OS X and it show stuff just after the query run out!

**Comment by krolow** [comment]
I think to begin a simple solution would be to add a refresh icon in the database, so we fetch the data once again.

It will not be automatic, but at least would give the support to refresh, and it might be a feature useful for other scenarios (like multiple users creating view/tables in same database) it would allow you to refresh and get the new things that others are generating...

And it also would be **simple**, and it will avoid user to have to close connection and reopen to get the new table/view/etc...

So I think it would be a good begin, afterwards we can check, some AST or another approach to handle those scenario where CREATE statements are generated through sqlectron...

What do you think guys?

**Comment by krolow** [comment]
may i give a go for it to implement a refresh icon what do you think @maxcnunes @BornaP ?

**Comment by BornaP** [comment]
Sure, give it a go! :)

**Comment by maxcnunes** [comment]
**TLDR** I guess will be possible to implement this auto reloading feature. I am working on it.

Since having a AST would give us some good analyse power over the executed queries. I tried find out some good sql-parser in node. But every parser I found is not fully implemented. Then I decided trying to create a bind from some existing one from C. But that would require creating a bind for each database client and some of them would not be possible because are private source (mssql). So my next attempt was writing a [sql parser](https://github.com/maxcnunes/sql-parser) my self in node. But after some hard working and fun I just realised would take some free time that I am not really interesting in investing right now. So my last attempt was writing a sql parser only for our main needs. Which is identify the query statement types (e.g. Insert, Select, CreateTable, and etc.). [sql-query-identifier](https://github.com/maxcnunes/sql-query-identifier) is the result of this idea.
The postgres client already returns the query identifier (I guess is because of that the Postico hava that feature to auto reload tables after creating them). So my main idea is add to sqlectron-core this new feature returning the query type for each executed statement. sql-query-identifier will be used as fallback for every database which does not return the statement type (I guess only postgres will not use it).

**Comment by BornaP** [comment]
Wow, that's some great news! The whole idea that you went on writing your own SQL parser is just impressive itself, and now that you have working solution that's just brilliant :) Bem feito @maxcnunes ! (hope I got it right haha)

**Comment by maxcnunes** [comment]
Closing this issue since have been implemented on #156 and #160.
